adele-booking/src/main/java/com/microservicesteam/adele/booking/domain/BookingRequest.java

```diff
@@ -2,9 +2,13 @@
 
 import java.util.List;
 
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import org.immutables.value.Value;
 
 @Value.Immutable
+@JsonSerialize(as = ImmutableBookingRequest.class)
+@JsonDeserialize(as = ImmutableBookingRequest.class)
```

Comment:
This is a sad addition to this type definition. Even though these annotations are added to `AdeleStyle`, the `as` part is crutial for Spring to be able to actually interpret the incoming booking request in JSON.
Not sure if there's a better, Immutables way to do this.